other_enrichment_methods/nbconverted/1_simulate_data.py

```diff
@@ -0,0 +1,219 @@
+)
+
+# Load metadata file with grouping assignments for samples
+metadata_filename = os.path.join(
+    base_dir,
+    dataset_name,
+    "data",
+    "metadata",
+    f"{project_id}_groups.tsv"
+)
+
+# Load pickled file
+with open(scaler_filename, "rb") as scaler_fh:
+    scaler = pickle.load(scaler_fh)
+
+
+# In[4]:
+
+
+# Output files
+gene_summary_filename = os.path.join(
+    base_dir, 
+    dataset_name, 
+    f"generic_gene_summary_{project_id}.tsv"
+)
+
+pathway_summary_filename = os.path.join(
+    base_dir, 
+    dataset_name, 
+    f"generic_pathway_summary_{project_id}.tsv"
+)
+
+
+# ### Simulate experiments using selected template experiment
+# 
+# Workflow:
+# 
+# 1. Get the gene expression data for the selected template experiment
+# 2. Encode this experiment into a latent space using the trained VAE model
+# 3. Linearly shift the encoded template experiment in the latent space
+# 4. Decode the samples. This results in a new experiment
+# 5. Repeat steps 1-4 to get multiple simulated experiments
+
+# In[5]:
+
+
+# Simulate multiple experiments
+# This step creates the following files in "<local_dir>/pseudo_experiment/" directory:           
+#   - selected_simulated_data_SRP012656_<n>.txt
+#   - selected_simulated_encoded_data_SRP012656_<n>.txt
+#   - template_normalized_data_SRP012656_test.txt
+# in which "<n>" is an integer in the range of [0, num_runs-1] 
+os.makedirs(os.path.join(local_dir, "pseudo_experiment"), exist_ok=True)
+for run_id in range(num_runs):
+    simulate_expression_data.shift_template_experiment(
+        normalized_compendium_filename,
+        project_id,
+        metadata_col_id,
+        NN_architecture,
+        dataset_name,
+        scaler,
+        local_dir,
+        base_dir,
+        run_id
+    )
+
+
+# ## Process template and simulated experiments
+# 
+# * Remove samples not required for comparison
+# * Make sure ordering of samples matches metadata for proper comparison
+# * Make sure values are cast as integers for using DESeq
+# * Filter lowly expressed genes for using DESeq
+
+# In[6]:
+
+
+if not os.path.exists(sample_id_metadata_filename):
+    sample_id_metadata_filename = None
+    
+stats.process_samples_for_DESeq(
+        mapped_template_filename,
+        metadata_filename,
+        processed_template_filename,
+        count_threshold,
+        sample_id_metadata_filename,
+    )
+
+for i in range(num_runs):
+    simulated_filename = os.path.join(
+        local_dir,
+        "pseudo_experiment",
+        f"selected_simulated_data_{project_id}_{i}.txt"
+    )
+    out_simulated_filename = os.path.join(
+        local_dir,
+        "pseudo_experiment",
+        f"selected_simulated_data_{project_id}_{i}_processed.txt"
+    )
+    stats.process_samples_for_DESeq(
+        simulated_filename,
+        metadata_filename,
+        out_simulated_filename,
+        count_threshold,
+        sample_id_metadata_filename,
+)
+
+
+# ### DELETE??? Differential expression analysis
```

Comment:
Do you want to delete this? It looks like you're using the results in `2_apply_enrichment_method.ipynb` (unless I'm mistaken).

Reply:
good catch.